chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,317 @@
+        assert peer is not None
+
+        assert isinstance(self.wallet_state_manager.observation_root, VaultRoot)
+        launcher_id = bytes32(self.wallet_state_manager.observation_root.launcher_id)
+
+        coin_states = await wallet_node.get_coin_state([launcher_id], peer)
+        if not coin_states:
+            raise ValueError(f"No coin found for launcher id: {launcher_id}.")
+        coin_state: CoinState = coin_states[0]
+        parent_state: CoinState = (await wallet_node.get_coin_state([coin_state.coin.parent_coin_info], peer))[0]
+        assert parent_state.spent_height is not None
+        launcher_spend = await fetch_coin_spend(uint32(parent_state.spent_height), parent_state.coin, peer)
+        launcher_solution = launcher_spend.solution.to_program()
+
+        is_recoverable = False
+        bls_pk = None
+        timelock = None
+        memos = launcher_solution.at("rrf")
+        secp_pk = memos.at("f").as_atom()
+        hidden_puzzle_hash = bytes32(memos.at("rf").as_atom())
+        if memos.list_len() == 4:
+            is_recoverable = True
+            bls_pk = G1Element.from_bytes(memos.at("rrf").as_atom())
+            timelock = uint64(memos.at("rrrf").as_int())
+            self.recovery_info = RecoveryInfo(bls_pk, timelock)
+        inner_puzzle_hash = get_vault_inner_puzzle_hash(
+            secp_pk, DEFAULT_CONSTANTS.GENESIS_CHALLENGE, hidden_puzzle_hash, bls_pk, timelock
+        )
+        vault_info = VaultInfo(
+            coin_state.coin,
+            launcher_id,
+            secp_pk,
+            hidden_puzzle_hash,
+            inner_puzzle_hash,
+            is_recoverable,
+            parent_state.coin.name(),
+        )
+
+        if coin_state.spent_height:
+            while coin_state.spent_height is not None:
+                coin_states = await wallet_node.fetch_children(coin_state.coin.name(), peer=peer)
+                odd_coin = None
+                for coin in coin_states:
+                    if coin.coin.amount % 2 == 1:
+                        if odd_coin is not None:
+                            raise ValueError("This is not a singleton, multiple children coins found.")
+                        odd_coin = coin
+                if odd_coin is None:
+                    raise ValueError("Cannot find child coin, please wait then retry.")
+                parent_state = coin_state
+                coin_state = odd_coin
+
+        vault_info = dataclasses.replace(vault_info, coin=coin_state.coin)
+        await self.save_info(vault_info)
+        await self.wallet_state_manager.create_more_puzzle_hashes()
+
+    async def save_info(self, vault_info: VaultInfo) -> None:
+        self.vault_info = vault_info
+        current_info = self.wallet_info
+        data_str = json.dumps(vault_info.to_json_dict())
+        wallet_info = WalletInfo(current_info.id, current_info.name, current_info.type, data_str)
+        self.wallet_info = wallet_info
```

Comment:
Let's not use the wallet info store here.  Maybe the singleton table used for DAOs in combination with another store for the vault wallet.